spec/classes/certs_apache_spec.rb

```diff
@@ -1,24 +1,28 @@
 require 'spec_helper'
 
 describe 'certs::apache' do
-  let :facts do
-    on_supported_os['redhat-7-x86_64']
-  end
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let :facts do
+        facts
```

Comment:
This is the old pattern that we still have all over the place. Recently the best practice has changed to:
```suggestion
  on_supported_os.each do |os, os_facts|
    context "on #{os}" do
      let(:facts) do
        os_facts
```

The benefit is that you don't have overlap between the variable and let block naming which makes things less confusing.